api/envoy/service/ext_proc/v3alpha/external_processor.proto

```diff
@@ -0,0 +1,307 @@
+  }
+}
+
+// For every ProcessingRequest received by the server with the "async_mode" field
+// set to false, the server must send back exactly one ProcessingResponse message.
+// [#next-free-field: 10]
+message ProcessingResponse {
+  oneof response {
+    option (validate.required) = true;
+
+    // The server must send back this message in response to a message with the
+    // "request_headers" field set.
+    HeadersResponse request_headers = 1;
+
+    // The server must send back this message in response to a message with the
+    // "response_headers" field set.
+    HeadersResponse response_headers = 2;
+
+    // The server must send back this message in response to a message with
+    // the "request_body" field set.
+    BodyResponse request_body = 3;
+
+    // The server must send back this message in response to a message with
+    // the "response_body" field set.
+    BodyResponse response_body = 4;
+
+    // The server must send back this message in response to a message with
+    // the "request_trailers" field set.
+    TrailersResponse request_trailers = 5;
+
+    // The server must send back this message in response to a message with
+    // the "response_trailers" field set.
+    TrailersResponse response_trailers = 6;
+
+    // If specified, attempt to create a locally generated response, send it
+    // downstream, and stop processing additional filters and ignore any
+    // additional messages received from the remote server for this request or
+    // response. If a response has already started -- for example, if this
+    // message is sent response to a "response_body" message -- then
+    // this will either ship the reply directly to the downstream codec,
+    // or reset the stream.
+    ImmediateResponse immediate_response = 7;
+  }
+
+  // Optional metadata that will be emitted as dynamic metadata to be consumed by the next
+  // filter. This metadata will be placed in the namespace "envoy.filters.http.ext_proc".
+  google.protobuf.Struct dynamic_metadata = 8;
+
+  // Override how parts of the HTTP request and response are processed
+  // for the duration of this particular request/response only. Servers
+  // may use this to intelligently control how requests are processed
+  // based on the headers and other metadata that they see.
+  envoy.extensions.filters.http.ext_proc.v3alpha.ProcessingMode mode_override = 9;
+}
+
+// The following are messages that are sent to the server.
+
+// This message is sent to the external server when the HTTP request and responses
+// are first received.
+message HttpHeaders {
+  // The HTTP request headers. All header keys will be
+  // lower-cased, because HTTP header keys are case-insensitive.
+  config.core.v3.HeaderMap headers = 1;
+
+  // The values of properties selected by the "request_properties"
+  // list in the configuration. Each entry in the map is the
+  // value of the specified property.
+  map<string, google.protobuf.Struct> properties = 2;
```

Comment:
FWIW, when https://github.com/envoyproxy/envoy/pull/13971 merges, it would be a good idea to add some changes explaining the ext_proc encoding for the properties (there is already verbiage there for CEL and Wasm).

Reply:
Makes sense to add the cross reference when it's available.

Reply:
It is now :)